### PR TITLE
[SPARK-41425][UI] Protobuf serializer for RDDStorageInfoWrapper

### PR DESCRIPTION
**Pull request by techaddict**
### What changes were proposed in this pull request?
Add Protobuf serializer for RDDStorageInfoWrapper

### Why are the changes needed?
Support fast and compact serialization/deserialization for RDDStorageInfoWrapper over RocksDB.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
New UT

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/status/protobuf/RDDStorageInfoWrapperSerializer.scala [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    new RDDStorageInfo(
+      id = info.getId,
+      name = info.getName,
+      numPartitions = info.getNumPartitions,
+      numCachedPartitions = info.getNumCachedPartitions,
+      storageLevel = info.getStorageLevel,
+      memoryUsed = info.getMemoryUsed,
+      diskUsed = info.getDiskUsed,
+      dataDistribution =
+        if (info.getDataDistributionList.isEmpty) {
```
Comment (techaddict) [inline core/src/main/scala/org/apache/spark/status/protobuf/RDDStorageInfoWrapperSerializer.scala:90]: this assignment is based on https://github.com/apache/spark/blob/d95fb4c33f6f061190fae091868117d182659147/core/src/main/scala/org/apache/spark/status/LiveEntity.scala#L645
dataDistribution is None for empty seq, and partitions is always Some() even when underlying collection is empty

**Comment by LuciferYang** [comment]
@techaddict Should change the pr title like `[SPARK-41425][CORE] Protobuf serializer for ...`, other pr titles need the same correction

**Comment by techaddict** [comment]
@LuciferYang I'm basing these PRs on #39048 #39096, which use this convention

**Comment by LuciferYang** [comment]
If so, we should add `[UI]`

**Comment by techaddict** [comment]
@gengliangwang merged latest master 👍🏼

File: core/src/main/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializer.scala [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 package org.apache.spark.status.protobuf
 
-import org.apache.spark.status.{ApplicationEnvironmentInfoWrapper, ApplicationInfoWrapper, JobDataWrapper, TaskDataWrapper}
+import org.apache.spark.status.{ApplicationEnvironmentInfoWrapper, ApplicationInfoWrapper, JobDataWrapper, RDDStorageInfoWrapper, TaskDataWrapper}
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializer.scala:20]: Let's change this to `org.apache.spark.status._` to save future troubles.

File: core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala [hunk 3]
```diff
@@ -21,8 +21,8 @@ import java.util.Date
 
 import org.apache.spark.{JobExecutionStatus, SparkFunSuite}
 import org.apache.spark.resource.{ExecutorResourceRequest, TaskResourceRequest}
-import org.apache.spark.status.{ApplicationEnvironmentInfoWrapper, ApplicationInfoWrapper, JobDataWrapper, TaskDataWrapper}
-import org.apache.spark.status.api.v1.{AccumulableInfo, ApplicationAttemptInfo, ApplicationEnvironmentInfo, ApplicationInfo, JobData, ResourceProfileInfo, RuntimeInfo}
+import org.apache.spark.status.{ApplicationEnvironmentInfoWrapper, ApplicationInfoWrapper, JobDataWrapper, RDDStorageInfoWrapper, TaskDataWrapper}
```
Comment (gengliangwang) [inline core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala:24]: Let's change this to `org.apache.spark.status._` to save future troubles.

File: core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala [hunk 4]
```diff
@@ -21,8 +21,8 @@ import java.util.Date
...
+import org.apache.spark.status.api.v1.{AccumulableInfo, ApplicationAttemptInfo, ApplicationEnvironmentInfo, ApplicationInfo, JobData, RDDDataDistribution, RDDPartitionInfo, RDDStorageInfo, ResourceProfileInfo, RuntimeInfo}
```
Comment (gengliangwang) [inline core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala:25]: Let's change this to `oorg.apache.spark.status.api.v1._` to save future troubles.

Comment (techaddict) [reply core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala]: done

File: core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala (hunk 3 above)
Comment (techaddict) [reply core/src/test/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializerSuite.scala]: done

File: core/src/main/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializer.scala (hunk 2 above)
Comment (techaddict) [reply core/src/main/scala/org/apache/spark/status/protobuf/KVStoreProtobufSerializer.scala]: done

**Comment by techaddict** [comment]
@gengliangwang addressed comments

File: core/src/main/scala/org/apache/spark/status/protobuf/RDDStorageInfoWrapperSerializer.scala (hunk 1 above)
Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/status/protobuf/RDDStorageInfoWrapperSerializer.scala:90]: Yeah the optional sequence here is tricky...

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
Thanks, merging to master

**Pull request closed without merging**